.github/workflows/test-go.yaml

```diff
@@ -65,6 +65,8 @@ jobs:
         echo "GO_TEST_TIMEOUT=1h" >> $GITHUB_ENV
 
     - name: Run Go Tests
+      # bash shell sets -eo pipefail so that the pipe to tee will exit with an error if test-go fails
```

Comment:
I don't think this is true. If you don't set `shell`, it defaults to `bash -e {0}`. See https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsshell. If you want to override it, I believe you need to set it to `bash {0}`. You can double check this by doing 

```
echo $SHELLOPTS 
```

Perhaps there is something else going wrong here?

Reply:
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#exit-codes-and-error-action-preference